src/AMSlib/wf/basedb.hpp

```diff
@@ -1339,15 +1471,18 @@ class RMQPublisherHandler : public AMQP::LibEventHandler
   /** @brief Number of messages successfully acknowledged */
   int _nb_msg_ack;
 
-  std::promise<ConnectionStatus> establish_connection;
-  std::future<ConnectionStatus> established;
+  std::promise<RMQConnectionStatus> establish_connection;
+  std::future<RMQConnectionStatus> established;
+
+  std::promise<RMQConnectionStatus> close_connection;
+  std::future<RMQConnectionStatus> closed;
 
-  std::promise<ConnectionStatus> close_connection;
-  std::future<ConnectionStatus> closed;
+  std::promise<RMQConnectionStatus> _error_connection;
+  std::future<RMQConnectionStatus> _ftr_error;
 
 public:
   std::mutex ptr_mutex;
-  std::vector<uint8_t*> data_ptrs;
+  std::vector<AMSMessage> data_ptrs;
```

Comment:
should we rename data_ptrs to `messages` or `nack_messages`?

Reply:
Yes, I have renamed it (and moved them to private)